src/puppeteer_pool.js

```diff
@@ -34,7 +34,7 @@ const DEFAULT_OPTIONS = {
 
 const mkdtempAsync = util.promisify(fs.mkdtemp);
 const rimrafAsync = util.promisify(rimraf);
-const DISK_CACHE_DIR = path.join(os.tmpdir(), 'puppeteer_disk_cache-');
+const DISK_CACHE_DIR = path.join(os.tmpdir(), `puppeteer_disk_cache-${Math.random() * 10000}`);
```

Comment:
I think the randomization is already there somewhere.

Reply:
I don't think so if you take a look at line 222 it calls `mkdtempAsync` with the `DISK_CACHE_DIR` constant directly

Reply:
Either way, it should not be connected to the target close error.

Reply:
The browser itself randomize the folder...